In app notifications

ChrisCruze opened the pull request:
Implemented In App Notifications after user Pushes on Goal including modal popup congratulating user

vivaromarf commented:
I ran it and it didn't seem to work right, the confetti did launch faster though.

vivaromarf commented:
Well it didn't work at all I didn't see a badge or notification in the ios simulator.

ChrisCruze commented:
> Well it didn't work at all I didn't see a badge or notification in the ios simulator.

Make sure to run the following command: 

`expo install`

vivaromarf commented:
<img width="542" alt="Screen Shot 2020-08-16 at 9 57 50 PM" src="https://user-images.githubusercontent.com/35068549/90350563-880c7000-e00b-11ea-8976-26041d2e4f59.png">

Got it to work thanks @ChrisCruze  Looks amazing so does the gear symbol and new pushable button @anandangalig

anandangalig commented on Components/Atoms/NotificationsState.js:
```diff
@@ -0,0 +1,142 @@
+import React, { useState } from "react";
```
do you need to import React and useState here? I don't see it being used here

ChrisCruze replied:
good catch

anandangalig commented on Components/Atoms/NotificationsState.js:
```diff
@@ -0,0 +1,142 @@
+
+const determineIfStreak = ({ streakCount }) => {
+  const is_divisible_by_5 = roundNumber(streakCount, 5) == streakCount;
+  return streakCount == 2 || is_divisible_by_5 & (streakCount > 2);
```
what does **&** do here? Is it same as &&?

anandangalig commented on Components/Atoms/NotificationsState.js:
```diff
@@ -0,0 +1,142 @@
+};
+
+const maxPushesText = ({ time_stamp_count }) => {
+  return `Avast Ye - ye just a new personal record of ${time_stamp_count} pushes for a single goal `;
```
ye just a new personal  - ye just **achieved** a new personal ?

ChrisCruze replied:
good catch

anandangalig commented:
Here are my thoughts:

1. Every single push has a popup, either smaller one up top or big one with a start. I think its too much. These types of things need to be mark a milestone and rendered every once in a while. 

2. 
![IMG_3971](https://user-images.githubusercontent.com/20729949/90839071-91a41980-e324-11ea-892c-41b8b37a0004.jpg)

3.
![IMG_3970](https://user-images.githubusercontent.com/20729949/90839087-9e287200-e324-11ea-96d2-a50ce5ea641a.jpg)

ChrisCruze commented:
> * Every single push has a popup, either smaller one up top or big one with a star. I think its too much. These types of things need to be marking a milestone and rendered every once in a while.
> *

The reason I like the small ones on top because I'd like to know my "score" immediately after I push, so I'd like to keep it unless we can come up with another solution that gives user immediate feedback after they push. I agree that the big one needs to come up less often though.

ChrisCruze commented:
I believe I addressed the majority of concerns here. Modal now only pops up when daily, weekly, monthly goals are complete. Also added a timer to it to make it better experience. In App Notification still pops up at the top though just so user can get immediate feedback
